.github/workflows/README.md

```diff
@@ -0,0 +1,115 @@
+# Workflow Taxonomy
+
+## Where and what tests are run
+
+---
+
+### Workflow: `deploy-happy-stack.yml`
+
+#### Ran on:
+
+- `deployment`: (deployments are triggered by `push-tests`)
+
+#### Jobs:
+
+- ##### `upgrade`: TODO
+- ##### `e2e-test`:
+  - Runs playwright against deployed frontend app uploading any artifacts
+    - ```bash
+            if [ "${DEPLOYMENT_STAGE}" == "stage" ]; then
+            export DEPLOYMENT_STAGE=staging
+          fi
+          if [ "${DEPLOYMENT_STAGE}" != "prod" ]; then
+            pip3 install -r scripts/smoke_tests/requirements.txt
+            python3 -m scripts.smoke_tests.setup
+          fi
+          cd frontend
+          npm ci
+          npx playwright install --with-deps
+          cp src/configs/${DEPLOYMENT_STAGE}.js src/configs/configs.js
+          DEBUG=pw:api npm run e2e-${DEPLOYMENT_STAGE}
+      ```
+- ##### `functional-test`:
+  - If environment is not `prod` start up backend container and run `make local-functional-test`
+  - `make local-functional-test` -> `python3 -m unittest discover --start-directory tests/functional --top-level-directory . --verbose`
+- ##### `performance-test`:
+  - If environment is `prod` run `make prod-performance-test`
+  - `make prod-performance-test` -> `python3 -m unittest discover --start-directory tests/performance --top-level-directory . --verbose`
+
+### Workflow: `lint-pr.yml`
+
+#### Ran on:
+
+- `pull_request_target`
+
+  - opening, editing, and syncing a pull request
+
+  #### Summary: Runs a GitHub action that lints the PR commit message
+
+### Workflow: `push-processing-base.yml`
+
+TODO
+
+### Workflow: `push-rdev.yml`
+
+TODO
+
+### Workflow: `push-tests.yml`
+
+#### Ran on:
+
+- `push` to `main`, `staging`, `prod`
+- `pull_request` targeting `"*"`
+
+#### Jobs:
+
+- ##### `lint`:
+
+  - Inside backend container: `make lint` -> `flake8 backend tests`
+  - Inside frontend container: (frontend/Makefile)`make lint` -> `npm run prettier-check & npm run lint`
+    - `npm run prettier-check` -> `prettier --check .`
+    - `npm run lint` -> `concurrently \"node_modules/.bin/next lint\" \"node_modules/.bin/stylelint --fix '**/*.{js,ts,tsx,css}'\" \"npm run type-check\"`
+      - `node_modules/.bin/next lint` -> runs eslint with config [next.js docs](https://nextjs.org/docs/basic-features/eslint)
+      - `node_modules/.bin/stylelint --fix '**/*.{js,ts,tsx,css}'` -> runs stylelint with config
+      - `npm run type-check` -> `tsc --noEmit`
+
+- ##### `e2e-test`:
+
+  - Installs dependencies and runs frontend server pointing at dev deployment:
+    - `npm ci &&npx playwright install --with-deps && cp src/configs/dev.js src/configs/configs.js && npm run dev&`
+  - Runs e2e tests:
+    - `npm run e2e` ->`playwright test`
+      - runs all tests in `frontend/tests` using playwright config, no TEST_ENV provided defaults to `local`
+  - Pushes images to RCS
```

Comment:
Why push images as part of the e2e-test step? Seems like an overextension.

Reply:
My understanding is that Infra helped us set up this way, so we only push images that pass the tests. And thus actual deployment will be shorter since the images will already be available in ECS (instead of pushing images at deployment time)

Reply:
This should probably be brought up to a larger visibility for the team: 

I think we can still push images in this workflow, but made it a separate run that relies on a positive e2e test run. Wouldn't that still meet that same functionality while allowing e2e tests to run quicker and potentially fail quicker?

Ideally, we'd refactor/shrink these workflow jobs to contain only the context and operations relative to a single operation (lint, functional test, smoke test, build, push, etc.)

Reply:
You mean updating the `if` in the `Push images` block to only push images when e2e tests succeed?

```
- name: Push images
        // Add extra condition to only push when success?
        if: (github.ref == 'refs/heads/main' || github.ref == 'refs/heads/staging' || github.ref == 'refs/heads/prod' ) 
        run: |
          echo "HAPPY_COMMIT=$(git rev-parse --verify HEAD)" >> envfile
          echo "HAPPY_BRANCH=$(git branch --show-current)" >> envfile
          happy push --docker-compose-env-file envfile --aws-profile "" --tag sha-${GITHUB_SHA:0:8} frontend backend
```

Reply:
it would be pulling Push images out of the `e2e-test` job and into its own job with a shape similar to:

```
push-happy-backend-frontend
    needs:
      - backend-unit-test
      - e2e-test
    runs-on: ubuntu-20.04
    if: (github.ref == 'refs/heads/main' || github.ref == 'refs/heads/staging' || github.ref == 'refs/heads/prod' )
    steps:
      ...
      - name: Push images
        run: |
          echo "HAPPY_COMMIT=$(git rev-parse --verify HEAD)" >> envfile
          echo "HAPPY_BRANCH=$(git branch --show-current)" >> envfile
          happy push --docker-compose-env-file envfile --aws-profile "" --tag sha-${GITHUB_SHA:0:8} frontend backend
       ...
```

While writing this out, I noticed that we're also pushing the backend image here.